addons/web/static/src/core/utils/hooks.js

```diff
@@ -48,18 +54,23 @@ export function useAutofocus({ refName, selectAll, mobile } = {}) {
         return ref;
     }
     // LEGACY
-    useEffect(
-        (el) => {
-            if (el && (!uiService.activeElement || uiService.activeElement.contains(el))) {
-                el.focus();
-                if (["INPUT", "TEXTAREA"].includes(el.tagName) && el.type !== "number") {
-                    el.selectionEnd = el.value.length;
-                    el.selectionStart = selectAll ? 0 : el.value.length;
-                }
+    let computeDependencies = () => [ref.el];
+    if (computeAdditionalDependencies) {
+        const _cpDep = computeDependencies;
+        computeDependencies = () => {
+            return _cpDep().concat(computeAdditionalDependencies());
+        };
+    }
+
```

Comment:
This feels overcomplicated, I think something like this should do the trick
```javascript
() => [ref.el].concat(computeAdditionalDependencies?.() ?? [])
```

Reply:
The point was to avoid evaluating the if at each execution of the function, Your writing is probably more readable though